editor/src/core/model/element-template-utils.ts

```diff
@@ -543,6 +544,92 @@ export function insertJSXElementChild(
   return insertChildAndDetails(updatedComponents, null, importsToAdd) // TODO is this wrapper type needed?
 }
 
+export function insertJSXElementChildren(
+  projectContents: ProjectContentTreeRoot,
+  targetParent: InsertionPath,
+  elementsToInsert: Array<JSXElementChild>,
+  components: Array<UtopiaJSXComponent>,
+  indexPosition: IndexPosition | null,
+): InsertChildAndDetails {
+  const parentPath: StaticElementPath = targetParent.intendedParentPath
+  let importsToAdd: Imports = {}
+  const updatedComponents = transformJSXComponentAtPath(components, parentPath, (parentElement) => {
+    if (isChildInsertionPath(targetParent)) {
+      if (!isJSXElementLike(parentElement)) {
+        throw new Error("Target parent for child element insertion doesn't support children")
+      }
+      let updatedChildren: Array<JSXElementChild>
+      if (indexPosition == null) {
+        updatedChildren = [...parentElement.children, ...elementsToInsert]
+      } else {
+        updatedChildren = elementsToInsert.reduce((working, elementToInsert) => {
+          return addToArrayAtIndexPosition(elementToInsert, working, indexPosition)
+        }, parentElement.children)
```

Comment:
I'm not sure this will handle all kinds of IndexPositions well

Reply:
fixed